ibis/backends/dask/execution/util.py

```diff
@@ -371,3 +458,36 @@ def _is_row_order_preserving(node: ops.Node):
             return (graph.proceed, True)
 
     return graph.traverse(_is_row_order_preserving, nodes)
+
+
+def rename_index(df: dd.DataFrame, new_index_name: str) -> dd.DataFrame:
+    # No elegant way to rename index
+    # https://github.com/dask/dask/issues/4950
+    df = df.map_partitions(pd.DataFrame.rename_axis, new_index_name, axis='index')
+    return df
+
+
+def ensure_divisions_known(df: dd.DataFrame | dd.Series) -> dd.DataFrame | dd.Series:
+    return_series = isinstance(df, dd.Series)
+
+    if df.known_divisions:
+        return df
+
+    if isinstance(df, dd.DataFrame) and 'index' in df:
+        # Special case where there's a column with name 'index'
+        unique_col_name = uuid.uuid4().hex
+
+        df = rename_index(df, unique_col_name)
+        df = df.reset_index().set_index(unique_col_name)
+        df = rename_index(df, None)
+    elif df.index.name is not None:
+        df = df.reset_index().set_index(df.index.name)
+    else:
+        df = df.reset_index().set_index('index')
```

Comment:
Every `set_index` here will force a computation which we want to avoid. You need to pass `divisons` and `sorted` if you want set index to be an actual lazy operation.

Reply:
Deleted this whole helper method.